src/components/common/Icon/Icon.jsx

```diff
@@ -0,0 +1,16 @@
+import { memo, lazy, Suspense } from 'react';
+import dynamicIconImports from 'lucide-react/dynamicIconImports';
+
+const fallback = <div style={{ background: '#ddd', width: 20, height: 20 }} />;
+
+const Icon = ({ name, className, ...props }) => {
```

Comment:
Could you rename it to `DynamicIcon`. This component should be used only for icons which names we are getting from the BE. In other case you should use common icons from lucid

https://lucide.dev/guide/packages/lucide-react#example

Reply:
The main reason is each time you use this component, the browser makes an extra request for icon. This is inefficient for icons whose names we know while developing